packages/wrangler/src/__tests__/publish.test.ts

```diff
@@ -1232,6 +1237,164 @@ export default{
         [32m%s[0m If you think this is a bug then please create an issue at https://github.com/cloudflare/wrangler2/issues/new."
       `);
     });
+
+    describe("expire assets", () => {
+      // it's a 100 seconds past epoch
+      // everyone's still talking about how great woodstock was
+      const DATE_NOW = 100000;
+      beforeEach(() => {
+        jest.spyOn(Date, "now").mockImplementation((): number => {
+          return DATE_NOW;
+        });
+      });
+      afterEach(() => {
+        (Date.now as jest.Mock).mockRestore();
+      });
+
+      it("should expire uploaded assets that aren't included anymore", async () => {
+        const assets = [
+          { filePath: "assets/file-1.txt", content: "Content of file-1" },
+          { filePath: "assets/file-2.txt", content: "Content of file-2" },
+        ];
+        const kvNamespace = {
+          title: "__test-name-workers_sites_assets",
+          id: "__test-name-workers_sites_assets-id",
+        };
+        writeWranglerToml({
+          main: "./index.js",
+          site: {
+            bucket: "assets",
+          },
+        });
+        writeWorkerSource();
+        writeAssets(assets);
+        mockUploadWorkerRequest();
+        mockSubDomainRequest();
+        mockListKVNamespacesRequest(kvNamespace);
+        mockKeyListRequest(kvNamespace.id, [
+          // Put file-1 in the KV namespace
+          { name: "assets/file-1.2ca234f380.txt" },
+          // As well as a couple from a previous upload
+          { name: "assets/file-3.somehash.txt" },
+          { name: "assets/file-4.anotherhash.txt" },
+        ]);
+
+        // Check that we pull down the values of file-3.txt and file-4.txt
+        setMockFetchKVGetValue(
+          "some-account-id",
+          "__test-name-workers_sites_assets-id",
+          "assets/file-3.somehash.txt",
+          btoa("Content of file-3")
```

Comment:
Strictly this should be

```suggestion
          Buffer.from("Content of file-3").toString('base64')
```

since `btoa()` is strongly deprecated.